### DISCUSSION TITLE
[Question]: 如何使用qwen2模型进行mlm任务训练

### DISCUSSION THREAD

**Discussion by lisi-lisi** [QwenLM/Qwen3#968] [Q&A]
### Has this been raised before?

- [X] I have checked [the GitHub README](https://github.com/QwenLM/Qwen2.5).
- [X] I have checked [the Qwen documentation](https://qwen.readthedocs.io) and cannot find an answer there.
- [X] I have searched [the issues](https://github.com/QwenLM/Qwen2.5/issues?q=is%3Aissue) and there is not a similar one.
- [X] I confirm that this is not a bug report, a feature request, or a badcase.

### Description

模仿bert模型进行mlm任务训练的代码，运行提示信息如下，如何使用qwen2模型进行mlm任务训练 ？
Traceback (most recent call last):
  File "D:\qwen2\mlm\mlm.py", line 12, in <module>
    model = AutoModelForMaskedLM.from_pretrained(model_name)
  File "D:\交通\.env\lib\site-packages\transformers\models\auto\auto_factory.py", line 567, in from_pretrained
    raise ValueError(
ValueError: Unrecognized configuration class <class 'transformers.models.qwen2.configuration_qwen2.Qwen2Config'> for this kind of AutoModel: AutoModelForMaskedLM.
Model type should be one of AlbertConfig, BartConfig, BertConfig, BigBirdConfig, CamembertConfig, ConvBertConfig, Data2VecTextConfig, DebertaConfig, DebertaV2Config, DistilBertConfig, ElectraConfig, ErnieConfig, EsmConfig, FlaubertConfig, FNetConfig, FunnelConfig, IBertConfig, LayoutLMConfig, LongformerConfig, LukeConfig, MBartConfig, MegaConfig, MegatronBertConfig, MobileBertConfig, MPNetConfig, MraConfig, MvpConfig, NezhaConfig, NystromformerConfig, PerceiverConfig, QDQBertConfig, ReformerConfig, RemBertConfig, RobertaConfig, RobertaPreLayerNormConfig, RoCBertConfig, RoFormerConfig, SqueezeBertConfig, TapasConfig, Wav2Vec2Config, XLMConfig, XLMRobertaConfig, XLMRobertaXLConfig, XmodConfig, YosoConfig.

**Comment by jklj077** [comment]
Hi, Qwen is Casual LM not Masked LM, so it cannot be used (or be recommeded to use) that way. The existence of casual mask renders it impossible to directly train with the objective of masked language modeling.

If you are using Qwen for sequence classification, consider using [Qwen2ForSequenceClassification](https://github.com/huggingface/transformers/blob/174890280b340b89c5bfa092f6b4fb0e2dc2d7fc/src/transformers/models/qwen2/modeling_qwen2.py#L1233).
